src/node/mod.rs

```diff
@@ -179,44 +179,70 @@ impl ChannelConfig {
     /// Represents an enumerated value describing the way channels must be matched between the
     /// node's inputs and outputs.
     pub(crate) fn count_mode(&self) -> ChannelCountMode {
-        self.inner.count_mode.load(Ordering::Acquire).into()
+        self.inner.lock().unwrap().count_mode
     }
 
-    fn set_count_mode(&self, v: ChannelCountMode) {
-        self.inner.count_mode.store(v as u32, Ordering::Release)
+    fn set_count_mode(&self, v: ChannelCountMode, registration: &AudioContextRegistration) {
+        self.inner.lock().unwrap().count_mode = v;
+
+        let message = ControlMessage::SetChannelCountMode {
+            id: registration.id(),
+            mode: v,
+        };
+        registration.context().send_control_msg(message).unwrap();
     }
 
     /// Represents an enumerated value describing the meaning of the channels. This interpretation
     /// will define how audio up-mixing and down-mixing will happen.
     pub(crate) fn interpretation(&self) -> ChannelInterpretation {
-        self.inner.interpretation.load(Ordering::Acquire).into()
+        self.inner.lock().unwrap().interpretation
     }
 
-    fn set_interpretation(&self, v: ChannelInterpretation) {
-        self.inner.interpretation.store(v as u32, Ordering::Release)
+    fn set_interpretation(
+        &self,
+        v: ChannelInterpretation,
+        registration: &AudioContextRegistration,
+    ) {
+        self.inner.lock().unwrap().interpretation = v;
```

Comment:
We need to hold on to this lock while sending the control message and drop the lock at the end of the method.
This prevent an inconsistent state when the thread is pre-empted after altering the `interpretation` and a concurrent call is made to this method.